core/husarnet/api_server/server.cpp

```diff
@@ -209,6 +211,25 @@ void ApiServer::runThread()
         returnSuccess(req, res);
       });
 
+  svr.Post(
+      "/api/claim", [&](const httplib::Request& req, httplib::Response& res) {
+        if(!validateSecret(req, res)) {
+          return;
+        }
+
+        if(proxy->isValid()) {
```

Comment:
This fill will quickly become overly repetitive if this check is left here (and more proxied requests added). Maybe add a function that'll wrap the logic here and produce an error (ideally returning it to the user too) internally. Can be a closure, can be a macro, can be a `if(method(args)){ return; }` construct

Reply:
Could I simply... Move the check inside `signAndForward`? The method already writes http responses back to user, might handle this class of errors too